frameit/lib/frameit/editor.rb

```diff
@@ -117,9 +117,19 @@ def complex_framing
         # Decrease the size of the framed screenshot to fit into the defined padding + background
         frame_width = background.width - horizontal_frame_padding * 2
         frame_height = background.height - top_space_above_device - vertical_frame_padding
-        @image.resize "#{frame_width}x"
-        if @image.height > frame_height
-          @image.resize "x#{frame_height.to_i}"
+
+        if fetch_config['show_complete_frame']
+          # calculate the final size of the screenshot to resize in one go
+          # it may be limited either by the width or height of the frame
+          image_aspect_ratio = @image.width.to_f / @image.height.to_f
+          image_width = [frame_width, @image.width].min
+          image_height = [frame_height, image_width / image_aspect_ratio].min
+          image_width = image_height * image_aspect_ratio
```

Comment:
I don't immediately get it: why is the height calculated from the width, and the width calculated from the height, here?